domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/Claim.java

```diff
@@ -162,6 +169,14 @@ public LocalDateTime getSettlementReachedAt() {
         return Optional.ofNullable(sealedClaimDocumentSelfPath);
     }
 
+    public BigDecimal getTotalAmountTillToday() {
```

Comment:
Looking at the code that had to be added to support this I think it would be cleaner if these get methods did the calculation and not the service methods,

i.e. things like this aren't nice at all:
https://github.com/hmcts/cmc-claim-store/pull/137/files#diff-55cdaa2fa8cc537e615c8ca0118170a5R98

and the copy method isn't ideal.

Thoughts?

Reply:
Yeah, I agree. I am more than open for suggestions.

I'd:
1. move calculations to some separate service.
2. leave `copy` method as it.

More unit tests and un-ignore one int test (that was failing).

Reply:
Agree that these getters could calculate values the same way as we used to do it in the frontend.

That would mean that `uk.gov.hmcts.cmc.claimstore.services.interest.InterestCalculationService#calculateInterest` becomes a dependency of `uk.gov.hmcts.cmc.domain.models.Claim` however we can accept it.

That would remove need for custom `copy` method.

Reply:
@timja-kainos I had a chat about this with Patryk yesterday and calculating in getter methods was his first thought, but that would make a dependency from the models module to the core module.

I'm not sure now how we intended to use these modules. If the models module is for keeping entities only then this dependency wouldn't be very nice.

But if we were to do it as DDD suggests, i.e. to keep the domain model layer separate, then it shouldn't be an issue. We would need to extract all domain related logic into the models module (or a separate domain focused one) though. Maybe not all in the first step, but that could be one way for us to proceed.

Any more thoughts @geek-soft ?

Reply:
I would be keen to keep calculator class (e.g. service) close to the model in case we need to extract domain classes to separate logical module.

I understand that keeping everything in the domain might prove to be difficult in the future and it is not necessarily what we would like to do especially when it requires some HTTP requests to external services e.g. to check UK bank holidays.

I think we will be iterating / improving this over the time, for now keeping total value calculation in the claim class makes sense.